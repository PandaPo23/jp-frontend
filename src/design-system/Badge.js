import system from '../utils/System';
import Text from './Text';

const Badge = system('Badge', {
  extend: Text,
  letterSpacing: 'caps',
  borderRadius: 'circle',
  px: 1,
  opacity: 0.8,
  fontSize: 3,
  textAlign: 'center',
  textTransform: 'uppercase',
});

export default Badge;
