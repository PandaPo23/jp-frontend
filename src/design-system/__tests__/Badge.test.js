import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Badge from '../Badge';

describe('Bagde component', () => {
  it('should render as badge', () => {
    const wrapper = mount(<Badge />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
