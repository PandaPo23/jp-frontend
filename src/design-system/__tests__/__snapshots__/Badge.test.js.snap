// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bagde component should render as badge 1`] = `
.emotion-0 {
  font-size: 20px;
  padding-left: 4px;
  padding-right: 4px;
  -webkit-letter-spacing: caps;
  -moz-letter-spacing: caps;
  -ms-letter-spacing: caps;
  letter-spacing: caps;
  text-align: center;
  border-radius: circle;
  opacity: 0.8;
}

<Badge
  borderRadius="circle"
  fontSize={3}
  letterSpacing="caps"
  opacity={0.8}
  px={1}
  textAlign="center"
  textTransform="uppercase"
>
  <div
    className="emotion-0"
  />
</Badge>
`;
